### PR TITLE
introduce ConfigPostProcessor extension

### PR DESCRIPTION
**Pull request by beiwei30**
**What this PR does**:

introduce ConfigPostProcessor extension so that user can customize configs for both ServiceConfig and ReferenceConfig further during deployment time. 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Review by AlexStocks — APPROVED** [review]

**Review by cityiron — APPROVED** [review]

File: config/interfaces/config_post_processor.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * limitations under the License.
+ */
+
+package interfaces
+
+// ConfigPostProcessor is an extension to give users a chance to customize configs against ReferenceConfig and
+// ServiceConfig during deployment time.
+type ConfigPostProcessor interface {
+	// PostProcessReferenceConfig customizes ReferenceConfig's params
+	PostProcessReferenceConfig(params *map[string]string)
```
Comment (watermelo) [inline config/interfaces/config_post_processor.go:24]: A map is just a pointer to a runtime.hmap structure, so we can replace`*map[string]string` -> `map[string]string`,.

Comment (beiwei30) [reply config/interfaces/config_post_processor.go]: got it, and fixed.

**Review by watermelo — APPROVED** [review]

**Comment by beiwei30** [comment]
@watermelo in order to give more flexibility to the implementation. I introduce an empty interface for the parameter instead of a parameter's type, since people may need to look into other infos like interface or others.

**Comment by beiwei30** [comment]
pls. hold on, I am still working on a better idea.

**Review by AlexStocks — APPROVED** [review]

**Comment by beiwei30** [comment]
@AlexStocks @watermelo @cityiron would you mind to review the new solution again?

**Review by hxmhlt — APPROVED** [review]

**Review by fangyincheng — APPROVED** [review]

**Review by georgehao — APPROVED** [review]

**Pull request merged**